src/Backups/RestoreCoordinationRemote.cpp

```diff
@@ -234,6 +235,46 @@ bool RestoreCoordinationRemote::acquireReplicatedSQLObjects(const String & loade
     return result;
 }
 
+bool RestoreCoordinationRemote::acquireInsertingDataForKeeperMap(const String & root_zk_path)
+{
+    bool result = false;
+    auto holder = with_retries.createRetriesControlHolder("acquireInsertingDataForKeeperMap");
+    holder.retries_ctl.retryLoop(
+        [&, &zk = holder.faulty_zookeeper]()
+        {
+            with_retries.renewZooKeeper(zk);
+
+            /// we need to remove leading '/' from root_zk_path
+            auto normalized_root_zk_path = std::string_view{root_zk_path}.substr(1);
+            std::string restore_lock_path = fs::path(zookeeper_path) / "keeper_map_tables" / normalized_root_zk_path / "restore_lock";
+            zk->createAncestors(restore_lock_path);
+            result = zk->tryCreate(restore_lock_path, "restorelock", zkutil::CreateMode::Persistent) == Coordination::Error::ZOK;
+
+            if (result)
+                return;
+
+            /// there can be an edge case where a path contains `/restore_lock/ in the middle of it
```

Comment:
You could use `escapeForFileName(root_zk_path)` and forget about that edge case.